docs: Add API and CLI documentation (#3917)
dagger/dagger: docs/current/162770-faq.md

Dagger SDKs use your existing Docker credentials and do not require separate authentication. Simply execute `docker login` against your container registry on the host where your Dagger pipelines are running.

> ### What API framework does Dagger use?

Reviewer: ```suggestion
### What API query language does Dagger use?
```

> Dagger uses GraphQL as its low-level language-agnostic API framework.

Reviewer: ```suggestion
Dagger uses GraphQL as its low-level language-agnostic API query language.
```

> ### Do I need to know GraphQL for Dagger?

Reviewer: ```suggestion
### Do I need to know GraphQL to use Dagger?
```

> No. You only need to know one of Dagger's supported languages to use Dagger. Communicating with the Dagger API is as simple as calling Dagger SDK functions in the language you choose to use. The conversion of those function calls to language-agnostic API calls is performed by the Dagger SDK. You do not need to be an expert in GraphQL to use Dagger.

Reviewer: ```suggestion
No. You only need to know one of Dagger's supported SDKs languages to use Dagger. The translation to underlying GraphQL  API calls  is handled internally by the Dagger SDK of your choice. 
```

### There's no SDK for &lt;language&gt; yet. Can I still use Dagger?

> Yes. It's possible to use the Dagger API from any language that [supports GraphQL](https://github.com/chentsulin/awesome-graphql) or from the Dagger CLI.

Reviewer: ```suggestion
Yes. It's possible to use the Dagger API from any language that [supports GraphQL](https://graphql.org/code/) or from the Dagger CLI.
```

Add a link to the `Dagger CLI` docs?
